src/wallet/wallet.cpp

```diff
@@ -2042,6 +2054,9 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int
         if (output.nDepth < (pcoin->IsFromMe(ISMINE_ALL) ? nConfMine : nConfTheirs))
             continue;
 
+        if (pcoin->InMempool() && std::max(pcoin->GetMempoolAncestorCount(), pcoin->GetMempoolDescendantCount()) >= nMaxAncestors)
```

Comment:
Given that we don't hold `mempool.cs` here, the transaction could be removed from the mempool in between the `pcoin->InMempool()` and `pcoin->GetMempoolAncestorCount()` calls, causing a segfault in the latter. Please abstract this whole conditional out into a method on `CTxMempool`, which takes the limit value and the txid, and returns whether the transaction is in the pool and satisfies the limits.